libpromises/promises.c

```diff
@@ -625,25 +625,40 @@ Promise *ExpandDeRefPromise(EvalContext *ctx, const Promise *pp, bool *excluded)
 
     /* Look for 'if'/'ifvarclass' exclusion. */
     {
+        /* We need to make sure to check both 'if' and 'ifvarclass' constraints. */
```

Comment:
since if is supposed to be an alias to ifvarclass, I don't think that we should apply both if they are on the same promise. Rather, I think we should disallow using both on a single promise.

Reply:
@nickanderson It's a little more subtle, since class guards with variables in them are implemented by putting them in an `if` contstraint. The easiest right now, would be to add a check, to ensure that you are only using 1 if/ifvarclass (in addition to the one added by class guard).

Reply:
> implemented by putting them in an if contstraint
Actually, they are put in `ifvarclass` constraint.

Reply:
@vpodzime but `ifvarclass` is an alias for `if` ;)